examples/panel/book-recommender/rag_book_recommender.py

```diff
@@ -0,0 +1,89 @@
+import os
+import json
+import pickle
+import pandas as pd
+
+from pathlib import Path
+from openai import OpenAI
+
+client = OpenAI()
+
+file_path = os.path.join('assets', 'goodreads.csv')
+df = pd.read_csv(file_path)
+
+
+file_path = os.path.join('assets', 'title_to_description.pkl')
+with open(file_path, 'rb') as file:
+    DESCRIPTIONS = pickle.load(file)
+
+
+def get_authors():
+    file_path = os.path.join('assets', 'author_to_title.pkl')
+    with open(file_path, 'rb') as file:
+        authors = pickle.load(file)
+    return authors
+
+
+def get_embeddings():
+    file_path = os.path.join('assets', 'embeddings.json')
+    with open(file_path, "r", encoding="utf-8") as file:
+        embeddings_json = json.load(file)
+    return embeddings_json
+
+
+def get_book_description_by_title(title):
+    return DESCRIPTIONS[title.upper()]
+
+
+class EmbeddingsStore:
+    def __init__(self):
+        self._path = Path("embeddings_another.json")
+
+        if not self._path.exists():
+            self._data = {}
+        else:
+            self._data = json.loads(self._path.read_text())
+
+    def get_one(self, text, title=None):
+        if text in self._data:
+            return self._data[text]
```

Comment:
I'm confused here

in line 83, you're doing:

```python
store.get_one(row["description"], row["title"])
```

I'm assuming description is a text describing the book and title is the book's title, this means that when get_one is executed, text=description...

then you check id the description is in `self._data`, and if so return the value (the embedding)

but then, in line 56, you do:  `self._data[title] = embedding`

meaning, you store the title as the key and the embedding as the value.

hence, line 48 will always be false, right?
```
        if text in self._data:
            return self._data[text]
```

because text (the description) is *never* in self._data, since you're storing the book's title, not the description

Reply:
I have replaced this logic with [this function](https://github.com/ploomber/doc/pull/127/files#diff-396e0215830b7ad98dc178d3fd362e3e0a577ab3509a784b915a8242e092be98R16-R34) since the `EmbeddingsStore` is being used for caching embeddings for dynamically generated embeddings. In this use-case we already have a static book dataset so I have pre-computed all the embeddings and stored in the file, so I don't think we need the caching mechanism? Is this assumption correct?

Reply:
yeah we can get rid of the EmbeddingsStor if you want

Reply:
Removed it already since now embeddings are being generated using [this function](https://github.com/ploomber/doc/pull/127/files#diff-396e0215830b7ad98dc178d3fd362e3e0a577ab3509a784b915a8242e092be98R16-R34)